shared/src/types/header_map/memory.rs

```diff
@@ -99,7 +101,7 @@ impl MemoryMap {
     pub(crate) fn remove(&self, key: &Byte32) -> Option<HeaderIndexView> {
         let mut guard = self.0.write();
         let ret = guard.remove(key);
-        shrink_to_fit!(guard, SHRINK_THRESHOLD);
+        // shrink_to_fit!(guard, SHRINK_THRESHOLD);
```

Comment:
why remove this？

Reply:
I accidentally commented out the line while investigating if the `SHRINK_THRESHOLD` could impact the performance of `header_map`. I have now reverted the change.

Reply:
![image](https://github.com/nervosnetwork/ckb/assets/46400566/637892aa-4be8-4f85-854f-efeed112e49e)
I believe this line significantly affects the performance of HeaderMap.

Reply:
@driftluo, Perhaps we should avoid calling `shrink_to_fit` during IBD, but we can call it during non-IBD periods.

Reply:
I think it's ok